assets/js/translator.js

```diff
@@ -65,13 +65,18 @@ if(modeEnabled('translation')) {
         }
 
         function setupWebpageTranslation() {
+            if(window.location.href.indexOf('&qP=') != -1) {
```

Comment:
You could call `translate` or the appropriate function at the end of `showTranslateWebpageInterface`? Also,  can we fix the other problem like this:

```
@@ -647,9 +648,11 @@ function translateText() {
         }
     }

-    if($('div#translateText').is(':visible')) {
+    var originalText = $('#originalText').val();
+
+    if($('div#translateText').is(':visible') && originalText.length) {
         if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
-            sendEvent('translator', 'translate', curSrcLang + '-' + curDstLang, $('#originalText').val().length);
+            sendEvent('translator', 'translate', curSrcLang + '-' + curDstLang, originalText.length);

             if(translateRequest) {
                 translateRequest.abort();
@@ -666,7 +669,7 @@ function translateText() {
                 request = {'langpair': curSrcLang + '|' + curDstLang};
             }

-            request.q = $('#originalText').val(); // eslint-disable-line id-length
+            request.q = originalText; // eslint-disable-line id-length
             request.markUnknown = $('#markUnknown').prop('checked') ? 'yes' : 'no';
             translateRequest = callApy({
                 data: request,
```